lms/resources/_js_config.py

```diff
@@ -60,6 +63,46 @@ def asdict(self):
         """
         return self._config
 
+    def enable_oauth_error_mode(
+        self, error_details, is_scope_invalid=False, requested_scopes=None
+    ):
+        """
+        Put the JavaScript code into "OAuth authorization failed" mode.
+
+        This mini-app is shown when OAuth authorization with Canvas fails
+        after redirecting to Canvas's OAuth authorization endpoint.
+
+        :param error_details: Technical details of the error
+        :type error_details: str
+        :param is_scope_invalid: `True` if authorization failed because the
+          OAuth client does not have access to all the necessary scopes
+        :type is_scope_invalid: bool
+        :param requested_scopes: List of Canvas API scopes that were requested
+        :type requested_scopes: list(str)
+        """
+        self._config.update(
+            {
+                "mode": "canvas-auth-error-dialog",
+                "invalidScope": is_scope_invalid,
+                "errorDetails": error_details,
+                "scopes": requested_scopes or [],
+            }
+        )
+
+    def enable_lti_launch_mode(self):
+        self._config["mode"] = "basic-lti-launch"
+
+        self._config["api"]["sync"] = self._sync_api()
+
+        # The config object for the Hypothesis client.
+        # Our JSON-RPC server passes this to the Hypothesis client over
+        # postMessage.
+        self._config["hypothesisClient"] = self._hypothesis_client
+
+        self._config["rpcServer"] = {
+            "allowedOrigins": self._request.registry.settings["rpc_allowed_origins"]
+        }
```

Comment:
I'm scratching my head about this one. Are these lines of code that would crash for oauth redirect error requests? Or could these settings actually remain the defaults without harm?